ops/pebble.py

```diff
@@ -3211,6 +3310,49 @@ def get_notices(
         resp = self._request('GET', '/v1/notices', query)
         return [Notice.from_dict(info) for info in resp['result']]
 
+    def get_identities(self) -> Dict[str, Identity]:
+        """Get all identities in Pebble.
+
+        Returns:
+            A dict mapping identity names to :class:`Identity` objects.
+
+        Raises:
+            :class:`ops.pebble.IdentityError`: If the server returns null for any identity.
+        """
+        resp = self._request('GET', '/v1/identities')
+        result = resp['result']
+
+        identities: Dict[str, Identity] = {}
+        for name, identity in result.items():
+            if identity is None:
+                raise IdentityError(f'server returned null identity {name!r}')
+            identities[name] = Identity.from_dict(identity)
+        return identities
```

Comment:
Seeing we're removing IdentityError, this can probably become a dict comprehension:
```suggestion
        return {name: Identity.from_dict(d) for name, d in result.items()}
```